core/src/test/java/org/elasticsearch/common/io/stream/AbstractWriteableEnumTestCase.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.io.stream;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Abstract class offering base functionality for testing @{link Writeable} enums.
+ */
+public abstract class AbstractWriteableEnumTestCase extends ESTestCase {
+
+    public void testReadFrom() throws IOException {
+        doReadFrom();
+    }
+
+    public void testWriteTo() throws IOException {
+        doWriteTo();
+    }
+
+    protected abstract void doReadFrom() throws IOException;
+
+    protected abstract void doWriteTo() throws IOException;
+
+    // a convenience method for testing the write of a writeable enum
+    protected void writeToStream(final Writeable writeableEnum, final int ordinal) throws IOException {
```

Comment:
Can we rename this to `assertWriteToStream` and the below method to `assertReadFromStream` so that it's clear that they both serialize and check the enum instead of just serialising it?